README.md

```diff
@@ -85,12 +117,415 @@ return (
+      showHint: ({preset, step}) => {/**/},
+      stepPass: ({preset, step}) => {/**/},
+      addPreset: ({preset}) => {/**/},
+      beforeRunPreset: ({preset}) => {/**/},
+      runPreset: ({preset}) => {/**/},
+      finishPreset: ({preset}) => {/**/},
+      beforeSuggestPreset: ({preset}) => {/**/},
+      beforeShowHint: ({stepData}) => {/**/},
+      stateChange: ({state}) => {/**/},
+      hintDataChanged: ({state}) => {/**/},
+      closeHint: ({hint}) => {/**/},
+      init: () => {/**/},
+      wizardStateChanged: ({wizardState}) => {/**/},
+    },
+  },
+};
+```
+
+---
+
+For default preset you can specify properties:
+
+```typescript jsx
+const onboardingOptions = {
+  config: {
+    presets: {
+      createProject: {
+        // preset name should be unique
+        name: 'Creating project', // text for user
+        type: 'default', // optional. 'default'(default value) | 'interlal' | 'combined'
+        visibility: 'visible', // optional. 'visible'(defaule value) | 'initialHidden' | 'alwaysHidden';
+        description: '', // optional, text for user
+        steps: [
+          {
+            slug: 'openBoard', // step slug must be unique across all presets
+            name: '', // text to show in popup
+            description: '', // text to show in popup
+            placement: 'top', // optional. Hint placement for step
+            passMode: 'onAction', // optional. 'onAction'(default value) | 'onShowHint' - trigger step pass on hint show
+            hintParams: {}, // optional. any custom properties for hint
+            closeOnElementUnmount: false, // optional. default valeue - false. Will close hint when element umnounts. 'True' not reccomended in general^ but may me helpful for some corners
+            passRestriction: 'afterPrevious', // optional. afterPrevious will block pass step is previous not passed
+            hooks: {
+              // optional
+              onStepPass: () => {/**/},
+              onCloseHint: () => {/**/},
+              onCloseHintByUser: () => {/**/},
+            },
+          },
+        ],
+        hooks: {
+          // optional
+          onBeforeStart: () => {/**/},
+          onStart: () => {/**/},
+          onEnd: () => {/**/},
+        },
+      },
+    },
+  },
+};
+```
+
+Also, there are combined preset. It is group presets, but it acts like one. When combined preset runs, it resolves to one of internal preset. You can find example in [test data](https://github.com/gravity-ui/onboarding/blob/main/src/tests/utils.ts#L71)
+
+## Plugins and event
+
+You can use event system. Available events: `showHint`, `stepPass`, `addPreset`, `beforeRunPreset`, `runPreset`, `finishPreset`, `beforeSuggestPreset`, `beforeShowHint`, `stateChange`, `hintDataChanged`, `closeHint`, `init`, `wizardStateChange`
+
+```typescript jsx
+controller.events.subscribe('beforeShowHint', callback);
+```
+
+---
+
+You can use plugins
+
+- MultiTabSyncPlugin - closes hint in all browser tabs
```

Comment:
It would be nice to add some description about plugins props and initializing arguments